src/rules/flake8_pie/rules.rs

```diff
@@ -106,6 +108,41 @@ pub fn dupe_class_field_definitions<'a, 'b>(
     }
 }
 
+/// PIE796
+pub fn prefer_unique_enums<'a, 'b>(checker: &mut Checker<'a>, parent: &'b Stmt, body: &'b [Stmt])
+where
+    'b: 'a,
+{
+    let StmtKind::ClassDef { bases, .. } = &parent.node else {
+        return;
+    };
+
+    if !bases.iter().any(|expr| {
+        checker
+            .resolve_call_path(expr)
```

Comment:
I tweaked this logic to use `resolve_call_path`, which validates that the expression is bound to `enum.Enum` (i.e., it tracks imports, follows aliases, and so on).